syntax = "proto3";

option go_package = "pb/game";

import "common/empty.proto";

import "common/common.proto";

import "common/common_message.proto";

option cc_generic_services = true;

message GsMessageExtern
{
  uint64 session_id = 1;
}

message GateNodeKickConnRequest
{
  uint64 session_id = 1;
}

message BroadCast2PlayerRequest
{
 repeated uint64 session_list = 1;
 MessageBody body = 2;
}

service GateService {
  rpc RegisterGame (RegisterGameNodeRequest) returns (Empty);
  rpc UnRegisterGame (UnRegisterGameNodeRequest) returns (Empty);
  rpc PlayerEnterGs (RegisterSessionGameNodeRequest) returns (RegisterSessionGameNodeResponse);
  rpc PlayerMessage (NodeRouteMessageRequest) returns (Empty);
  rpc KickConnByCentre (GateNodeKickConnRequest) returns (Empty);
  rpc RouteNodeStringMsg (RouteMsgStringRequest) returns (RouteMsgStringResponse);
  rpc RoutePlayerStringMsg (RoutePlayerMsgStringRequest) returns (RoutePlayerMsgStringResponse);
  rpc BroadCast2PlayerMessage (BroadCast2PlayerRequest) returns (Empty);
}